pkg/products/grafana/reconciler.go

```diff
@@ -0,0 +1,314 @@
+		return phase, err
+	}
+
+	phase, err = r.reconcileSubscription(ctx, client, installation, productNamespace, operatorNamespace)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, fmt.Sprintf("Failed to reconcile %s subscription", constants.ThreeScaleSubscriptionName), err)
+		return phase, err
+	}
+
+	phase, err = r.reconcileComponents(ctx, client, installation)
+	if err != nil || phase != integreatlyv1alpha1.PhaseCompleted {
+		events.HandleError(r.recorder, installation, phase, "Failed to create components", err)
+		return phase, err
+	}
+
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+func (r *Reconciler) reconcileSecrets(ctx context.Context, client k8sclient.Client, installation *integreatlyv1alpha1.RHMI, cr *grafanav1alpha1.Grafana) (integreatlyv1alpha1.StatusPhase, error) {
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "grafana-k8s-proxy",
+			Namespace: r.Config.GetOperatorNamespace(),
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(ctx, client, secret, func() error {
+		owner.AddIntegreatlyOwnerAnnotations(secret, installation)
+		if secret.Data == nil {
+			secret.Data = map[string][]byte{}
+		}
+		secret.Data["session_secret"] = []byte(populateSessionProxySecret())
+		return nil
+	})
+
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	return integreatlyv1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) reconcileComponents(ctx context.Context, client k8sclient.Client, installation *integreatlyv1alpha1.RHMI) (integreatlyv1alpha1.StatusPhase, error) {
+	r.logger.Info("reconciling grafana custom resource")
+
+	var annotations = map[string]string{}
+	annotations["service.alpha.openshift.io/serving-cert-secret-name"] = "grafana-k8s-tls"
+
+	var serviceAccountAnnotations = map[string]string{}
+	serviceAccountAnnotations["serviceaccounts.openshift.io/oauth-redirectreference.primary"] = "{\"kind\":\"OAuthRedirectReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Route\",\"name\":\"grafana-route\"}}"
+
+	grafana := &grafanav1alpha1.Grafana{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "grafana",
+			Namespace: r.Config.GetOperatorNamespace(),
+		}, Spec: grafanav1alpha1.GrafanaSpec{
+			Config: grafanav1alpha1.GrafanaConfig{
+				Log: &grafanav1alpha1.GrafanaConfigLog{
+					Mode:  "console",
+					Level: "warn",
+				},
+				Auth: &grafanav1alpha1.GrafanaConfigAuth{
+					DisableLoginForm:   &[]bool{false}[0],
+					DisableSignoutMenu: &[]bool{true}[0],
+				},
+				AuthBasic: &grafanav1alpha1.GrafanaConfigAuthBasic{
+					Enabled: &[]bool{true}[0],
+				},
+				AuthAnonymous: &grafanav1alpha1.GrafanaConfigAuthAnonymous{
+					Enabled: &[]bool{true}[0],
+				},
+			},
+			Containers: []v1.Container{
+				{Name: "grafana-proxy",
+					Image: "quay.io/openshift/origin-oauth-proxy:4.2",
```

Comment:
Do we hardcode the oauth proxy image elsewhere? I'm wondering if this is something that could be more configured.

Reply:
I took that from the Kafka project but I think we could set it as a parameter at least. Or we could set it to the latest tag but I dont know what the impact might be on changes.